atomicapp/display.py

```diff
@@ -0,0 +1,248 @@
+    initalization and set's the appropriate log level based on either the default
+    or custom level value.
+
+    Default python level codes
+
+    NOTSET    0
+    DEBUG    10
+    INFO     20
+    WARNING  30
+    ERROR    40
+    CRITICAL 50
+
+    Custom log levels in constants.py
+
+    LOG_LEVELS = {
+        "default": 90,
+        "cockpit": 91,
+        "stdout": 92,
+        "syslog": 93,
+        "none": 94
+    }
+
+    '''
+
+    # Console colour codes
+    codeCodes = {
+        'black': '0;30', 'bright gray': '0;37',
+        'blue': '0;34', 'white': '1;37',
+        'green': '0;32', 'bright blue': '1;34',
+        'cyan': '0;36', 'bright green': '1;32',
+        'red': '0;31', 'bright cyan': '1;36',
+        'purple': '0;35', 'bright red': '1;31',
+        'yellow': '0;33', 'bright purple': '1;35',
+        'dark gray': '1;30', 'bright yellow': '1;33',
+        'normal': '0'
+    }
+
+    def __init__(self):
+        """
+        Upon initialization of the Display class we grab the log level from
+        logger as well as the standard verbose level (ex. if -v was passed or not)
+        """
+        self.logger = logging.getLogger(LOG_NAME)
+        self.verbose_level = self.logger.getEffectiveLevel()
+        self.log_level = logging.getLogger(LOG_CUSTOM_NAME).getEffectiveLevel()
+
+    def debug(self, msg, only=None):
+        self._display(msg, LOG_LEVELS["debug"], 'cyan', only)
+
+    def verbose(self, msg, only=None):
+        self._display(msg, LOG_LEVELS["debug"], 'cyan', only)
+
+    def info(self, msg, only=None):
+        self._display(msg, LOG_LEVELS["info"], 'white', only)
+
+    def warning(self, msg, only=None):
+        self._display(msg, LOG_LEVELS["warning"], 'yellow', only)
+
+    def error(self, msg, only=None):
+        self._display(msg, LOG_LEVELS["error"], 'red', only)
+
+    def _display(self, msg, code, color, only):
+        """
+        Display checks to see what log_level is being matched to and passes it
+        along to the correct logging provider. If an unknown error occurs
+        retrieving the log level, we error out.
+        """
+        if self.log_level == LOG_LEVELS['stdout']:
+            self._stdout(msg, code)
+        elif self.log_level == LOG_LEVELS['cockpit']:
+            self._cockpit(msg, code, only)
+        elif self.log_level == LOG_LEVELS['syslog']:
+            self._syslog(msg, code)
+        elif self.log_level == LOG_LEVELS['none']:
+            return
+        else:
+            self._stdout_via_logger(msg, code, color)
+
+    def _stdout(self, msg, code):
+        """
+        Colorless logging output using the standard [DEBUG], [WARNING], etc tags.
+        """
+        if self.verbose_level is not LOG_LEVELS['debug'] and code is LOG_LEVELS['debug']:
+            return
+
+        if code == LOG_LEVELS['debug']:
+            msg = "[DEBUG] %6d %0.2f %s" % (os.getpid(), time.time(), msg)
```

Comment:
After running it a couple of times I feel like the extra output in the debug statements (the pid/time) doesn't really help debug anything any more. We don't run multiprocess so I don't think this is useful. Thoughts?